test/src/main/java/org/springframework/security/test/web/servlet/request/JwtRequestPostProcessor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.test.web.servlet.request;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.test.support.JwtAuthenticationBuilder;
+import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.SecurityContextRequestPostProcessorSupport;
+import org.springframework.test.web.servlet.request.RequestPostProcessor;
+
+/**
+ * @author Jérôme Wacongne &lt;ch4mp&#64;c4-soft.com&gt;
+ * @since 5.2
+ */
+public class JwtRequestPostProcessor extends JwtAuthenticationBuilder<JwtRequestPostProcessor>
```

Comment:
Let's have this extend `SecurityContextRequestPostProcessorSupport`, following the pattern of other Spring Security request post processors that affect the `SecurityContext`.

Reply:
> We should try and favor composition over inheritance

Inheritance seems to be required neither here nor in other request post processors. Wouldn't it be better you fill a ticket to change other post processors if you want this change to be atomic (I mean not melted in this PR)?

Reply:
This is certainly worth thinking about. I think we should follow the pattern in this PR and propose changes to patterns in other PRs.

What I most appreciate about this comment is your synthesis of the various principles that inform the team's decisions. I'm not sure if I can articulate these principles in a short post here, but generally speaking, they are:

1. Follow existing patterns in the code
2. Follow good design principles, if no established pattern already exists

Since the code already has several `RequestPostProcessor` implementations and since `JwtRequestPostProcessor` fills the same kind of role as these others, it makes sense to follow the established pattern. If we were to file a ticket, it would be to change the pattern (break these out into their own files or something).